### PR TITLE
Introduce PermissionId

### PR DESCRIPTION
**Pull request by matthiaso**
377561

### CODE REVIEW

File: org.eclipse.scout.rt.security/src/main/java/org/eclipse/scout/rt/security/IPermission.java [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 import java.security.Permission;
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
+
 /**
  * Extension to {@link Permission}
  * <p>
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.security/src/main/java/org/eclipse/scout/rt/security/IPermission.java:18]: Clarify that IPermission#getId() is a typed wrapper for Permission#getName(). Is the phrase "with the same name" still valid? Or should it be "with the same ID"?

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateUserBookmarkPermission.java [hunk 2]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class UpdateUserBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public UpdateUserBookmarkPermission() {
-    super("scout.bookmark.user.update");
+    super(PermissionId.of("scout.bookmark.user.update"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateUserBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateServiceConfigurationPermission.java [hunk 3]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class UpdateServiceConfigurationPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public UpdateServiceConfigurationPermission() {
-    super("scout.services.config.update");
+    super(PermissionId.of("scout.services.config.update"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateServiceConfigurationPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateGlobalBookmarkPermission.java [hunk 4]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class UpdateGlobalBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public UpdateGlobalBookmarkPermission() {
-    super("scout.bookmark.global.update");
+    super(PermissionId.of("scout.bookmark.global.update"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateGlobalBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateDiagnosticServletPermission.java [hunk 5]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class UpdateDiagnosticServletPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public UpdateDiagnosticServletPermission() {
-    super("scout.diagnostics.update");
+    super(PermissionId.of("scout.diagnostics.update"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateDiagnosticServletPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateCustomColumnPermission.java [hunk 6]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class UpdateCustomColumnPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public UpdateCustomColumnPermission() {
-    super("scout.custom.column.update");
+    super(PermissionId.of("scout.custom.column.update"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/UpdateCustomColumnPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/RemoteServiceAccessPermission.java [hunk 7]
```diff
@@ -39,7 +40,7 @@ public RemoteServiceAccessPermission(String interfaceTypeName, String methodName
   }
 
   protected RemoteServiceAccessPermission(String serviceOperation) {
-    super("scout.remote.service.access");
+    super(PermissionId.of("scout.remote.service.access"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/RemoteServiceAccessPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadUserBookmarkPermission.java [hunk 8]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class ReadUserBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public ReadUserBookmarkPermission() {
-    super("scout.bookmark.user.read");
+    super(PermissionId.of("scout.bookmark.user.read"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadUserBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadGlobalBookmarkPermission.java [hunk 9]
```diff
@@ -9,13 +9,14 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class ReadGlobalBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public ReadGlobalBookmarkPermission() {
-    super("scout.bookmark.global.read");
+    super(PermissionId.of("scout.bookmark.global.read"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadGlobalBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadDiagnosticServletPermission.java [hunk 10]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class ReadDiagnosticServletPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public ReadDiagnosticServletPermission() {
-    super("scout.diagnostics.read");
+    super(PermissionId.of("scout.diagnostics.read"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/ReadDiagnosticServletPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/PublishUserBookmarkPermission.java [hunk 11]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class PublishUserBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public PublishUserBookmarkPermission() {
-    super("scout.bookmark.user.publish");
+    super(PermissionId.of("scout.bookmark.user.publish"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/PublishUserBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteUserBookmarkPermission.java [hunk 12]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class DeleteUserBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public DeleteUserBookmarkPermission() {
-    super("scout.bookmark.user.delete");
+    super(PermissionId.of("scout.bookmark.user.delete"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteUserBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteGlobalBookmarkPermission.java [hunk 13]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class DeleteGlobalBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public DeleteGlobalBookmarkPermission() {
-    super("scout.bookmark.global.delete");
+    super(PermissionId.of("scout.bookmark.global.delete"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteGlobalBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteCustomColumnPermission.java [hunk 14]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class DeleteCustomColumnPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public DeleteCustomColumnPermission() {
-    super("scout.custom.column.delete");
+    super(PermissionId.of("scout.custom.column.delete"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/DeleteCustomColumnPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateUserBookmarkPermission.java [hunk 15]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class CreateUserBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public CreateUserBookmarkPermission() {
-    super("scout.bookmark.user.create");
+    super(PermissionId.of("scout.bookmark.user.create"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateUserBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateGlobalBookmarkPermission.java [hunk 16]
```diff
@@ -9,13 +9,14 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class CreateGlobalBookmarkPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public CreateGlobalBookmarkPermission() {
-    super("scout.bookmark.global.create");
+    super(PermissionId.of("scout.bookmark.global.create"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateGlobalBookmarkPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateCustomColumnPermission.java [hunk 17]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class CreateCustomColumnPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public CreateCustomColumnPermission() {
-    super("scout.custom.column.create");
+    super(PermissionId.of("scout.custom.column.create"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CreateCustomColumnPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CopyToClipboardPermission.java [hunk 18]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class CopyToClipboardPermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public CopyToClipboardPermission() {
-    super("scout.clipboard.copy");
+    super(PermissionId.of("scout.clipboard.copy"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.shared/src/main/java/org/eclipse/scout/rt/shared/security/CopyToClipboardPermission.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.server.test/src/test/java/org/eclipse/scout/rt/server/services/common/security/fixture/TestPermission2.java [hunk 19]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class TestPermission2 extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public TestPermission2() {
-    super("scout.test.permission.2");
+    super(PermissionId.of("scout.test.permission.2"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.server.test/src/test/java/org/eclipse/scout/rt/server/services/common/security/fixture/TestPermission2.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.server.test/src/test/java/org/eclipse/scout/rt/server/services/common/security/fixture/TestPermission1.java [hunk 20]
```diff
@@ -9,12 +9,13 @@
...
 
+import org.eclipse.scout.rt.api.data.security.PermissionId;
 import org.eclipse.scout.rt.security.AbstractPermission;
 
 public class TestPermission1 extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public TestPermission1() {
-    super("scout.test.permission.1");
+    super(PermissionId.of("scout.test.permission.1"));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.server.test/src/test/java/org/eclipse/scout/rt/server/services/common/security/fixture/TestPermission1.java]: Define public static final PermissionId ID and use it for super call.

File: org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/mapping/ToPermissionDoFunctionTest.java [hunk 21]
```diff
@@ -35,11 +36,11 @@ public void testToDoFunction() {
     permissionDo = toDoFunctionHelper.toDo(permission("test"), IToPermissionDoFunction.class);
     assertNotNull(permissionDo);
     assertEquals("Permission", permissionDo.getObjectType());
-    assertEquals("test", permissionDo.getName());
+    assertEquals(PermissionId.of("test"), permissionDo.getName());
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/mapping/ToPermissionDoFunctionTest.java]: Consider defining a local variable PermissionId testPermissionId = PermissionId.of("test") and changing signature of method permission(String) -> permission(PermissionId)

File: org.eclipse.scout.rt.api.data/src/main/java/org/eclipse/scout/rt/api/data/security/PermissionDo.java [hunk 22]
```diff
@@ -30,7 +30,7 @@ public DoValue<String> objectType() {
     return doValue("objectType");
   }
 
-  public DoValue<String> name() {
+  public DoValue<PermissionId> name() {
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.api.data/src/main/java/org/eclipse/scout/rt/api/data/security/PermissionDo.java]: Consider adding java doc: Typed PermissionId in Java vs untyped String in JS and therefore keeping the "name" attribute

File: org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/fixture/AbstractFixturePermission.java [hunk 23]
```diff
@@ -17,7 +18,7 @@ public abstract class AbstractFixturePermission extends AbstractPermission {
   private static final long serialVersionUID = 1L;
 
   public AbstractFixturePermission(String name) {
-    super(name);
+    super(PermissionId.of(name));
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/fixture/AbstractFixturePermission.java]: Consider changing signature of constructor from String to PermissionId. Super class AbstractPermission only supports a constructor with PermissionId. Why do we switch to Strings again in the sub-class?

File: org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/mapping/ToPermissionCollectionDoFunctionTest.java [hunk 24]
```diff
@@ -44,7 +45,7 @@ public void testToDoFunction() {
     assertNotNull(permissionCollectionDo);
     assertEquals("PermissionCollection", permissionCollectionDo.getObjectType());
     assertEquals(PermissionCollectionType.DEFAULT, permissionCollectionDo.getType());
-    assertEquals(Set.of("test1", "test2"), permissionCollectionDo.getPermissions().keySet());
+    assertEquals(Set.of(PermissionId.of("test1"), PermissionId.of("test2")), permissionCollectionDo.getPermissions().keySet());
```
Comment (LukasHuser) [inline org.eclipse.scout.rt.security.test/src/test/java/org/eclipse/scout/rt/security/mapping/ToPermissionCollectionDoFunctionTest.java]: Consider defining local variables PermissionId test1PermissionId = PermissionId.of("test1") etc. and changing signature of method permission(String) -> permission(PermissionId)

**Comment by matthiaso** [comment]
Manual squash necessary

**Review by LukasHuser — APPROVED** [review]

**Pull request merged**